crates/ide/src/references/rename.rs

```diff
@@ -20,17 +20,51 @@ use crate::{
     SourceChange, SourceFileEdit, TextRange, TextSize,
 };
 
+#[derive(Debug)]
+pub struct RenameError(pub(crate) String);
+
+impl RenameError {
+    fn new(message: &str) -> RenameError {
+        RenameError(String::from(message))
+    }
+}
+
+impl Display for RenameError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
```

Comment:
Minor, but we generally import `fmt`: https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#import-style